react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      current: 0,
+      num: [0, 1, 2, 3, 4]
+    };
+    this.position = this.position.bind(this);
+  };
+  componentDidMount() {
+    this.intervalId = setInterval(() => {
+      if(this.state.current === 4) {
+        this.setState({ current: 0 })
+        return;
+        }
+        this.setState({ current: this.state.current + 1})
+    }, 3000);
+  };
+  position() {
+    for (let i = this.state.current; i < this.state.num.length; i++) {
+      if (event.target.id === "left" && this.state.current !== 0) {
+        this.setState({current: this.state.current - 1});
+      } else if (event.target.id === "left" && this.state.current === 0) {
+        this.setState({current: 4})
+      } else if (event.target.id === "right" && this.state.current !== 4) {
+        this.setState({current: this.state.current + 1});
+      } else if (event.target.id === "right" && this.state.current === 4) {
+        this.setState({current: 0});
+      } else if (event.target.id !== "left" && event.target.id !== "right"){
+        this.setState({current: parseInt(event.target.id)});
+      }
+    }
```

Comment:
Instead of having 1 complex click handler you should make 3 simple click handlers. IE: `goToPreviousImage`, goToNextImage`, `goToImage(index)`. You can then directly place the appropriate click handler on the element that you want to perform that action.